lib/rubocop/cop/rspec/be_empty.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Prefer using `be_empty` when checking for an empty array.
+      #
+      # @example
+      #   # bad
+      #   expect(array).to contain_exactly
+      #   expect(array).to match_array([])
+      #
+      #   # good
+      #   expect(array).to be_empty
+      #
+      class BeEmpty < Base
+        extend AutoCorrector
+
+        MSG = 'Use `be_empty` matchers for checking an empty array.'
+        RESTRICT_ON_SEND = %i[contain_exactly match_array].freeze
+
+        # @!method expect_array_matcher?(node)
+        def_node_matcher :expect_array_matcher?, <<~PATTERN
+          (send
+            (send nil? :expect _)
+            #Runners.all
+            ${
+              (send nil? {:eq :eql :match_array} (array))
```

Comment:
It looks like it would be an offense for `eq([])` and `eql([])` as well, but what do you think?

```suggestion
              (send nil? :match_array (array))
```

Reply:
Fixed